crates/runc-shim/src/task.rs

```diff
@@ -93,13 +119,29 @@ impl<F, C> TaskService<F, C> {
             .await
             .unwrap_or_else(|e| warn!("send {} to publisher: {}", topic, e));
     }
+
+    fn run_oom_monitor(&self, mut rx: Receiver<String>, id: String) {
```

Comment:
Sorry, I think your previous commit is better: https://github.com/containerd/rust-extensions/commit/9d1ccc9fc8e5998b562130c6cc3e03d91a4cf72b#diff-91a6ff7d9d978824549619245220f7b4913562d009ff1c7a78502141d063421bR113, `send_event()` need execute in another thread but cloning the whole TaskService is not smart.

Reply:
Yes, I also agree with this. I will revert this part.